ergo-lib/src/wallet/multi_sig.rs

```diff
@@ -147,26 +147,30 @@ pub fn bag_for_multi_sig(
                 let simulated_found = simulated_propositions.contains(&leaf.proposition());
                 if real_found || simulated_found {
                     if real_found {
-                        let real_commitment: Hint =
-                            Hint::CommitmentHint(CommitmentHint::RealCommitment(RealCommitment {
-                                image: leaf.proposition(),
-                                commitment: leaf.commitment(),
-                                position: position.clone(),
-                            }));
-                        let real_secret_proof: Hint =
-                            Hint::SecretProven(SecretProven::RealSecretProof(RealSecretProof {
-                                image: leaf.proposition(),
-                                challenge: leaf.challenge(),
-                                unchecked_tree: UncheckedTree::UncheckedLeaf(leaf),
-                                position,
-                            }));
-                        bag.add_hint(real_commitment);
-                        bag.add_hint(real_secret_proof);
-                    } else {
+                        if let Some(commitment) = leaf.commitment_opt() {
+                            let real_commitment: Hint = Hint::CommitmentHint(
+                                CommitmentHint::RealCommitment(RealCommitment {
+                                    image: leaf.proposition(),
+                                    commitment,
+                                    position: position.clone(),
+                                }),
+                            );
+                            let real_secret_proof: Hint = Hint::SecretProven(
+                                SecretProven::RealSecretProof(RealSecretProof {
+                                    image: leaf.proposition(),
+                                    challenge: leaf.challenge(),
+                                    unchecked_tree: UncheckedTree::UncheckedLeaf(leaf),
+                                    position,
+                                }),
+                            );
+                            bag.add_hint(real_commitment);
+                            bag.add_hint(real_secret_proof);
+                        }
```

Comment:
In case of `leaf.commitment_opt()` is None we should throw an error. And in the case below as well. I suggest moving commitment extraction above `if real_found {...} else {...}` expression to avoid the repeat.

Reply:
Yes i moving commitment extraction to the above of `if` block